driver-core/src/main/com/mongodb/client/model/expressions/MqlExpression.java

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright 2008-present MongoDB, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.mongodb.client.model.expressions;
+
+import org.bson.BsonArray;
+import org.bson.BsonDocument;
+import org.bson.BsonString;
+import org.bson.BsonValue;
+import org.bson.codecs.configuration.CodecRegistry;
+
+import java.util.function.BinaryOperator;
+import java.util.function.Function;
+
+final class MqlExpression<T extends Expression>
```

Comment:
Check all access modifiers.

Reply:
Moved from [here](https://github.com/mongodb/mongo-java-driver/pull/1050#discussion_r1030858270): `MqlExpression.isBoolean` and similar methods should probably be `private`.

Reply:
- `MqlExpression.sort` should probably be package-access.
- `MqlExpression.reduce`/`isDate`/`isDocument` should probably be `private`.

Reply:
(Some of those are used by branches, so non-private)